armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -267,32 +301,95 @@ def axiallyExpandAssembly(self):
             for c in getSolidComponents(b):
                 c.clearCache()
+                postExpMass = c.getMass()
+                materialMasses[c.material.name] += postExpMass
+                componentMasses[c.name] += postExpMass
+                self.detailedMassConservationReport["(post) mass"].append(postExpMass)
+                self.detailedMassConservationReport["(post - pre) mass"].append(
+                    postExpMass - preMass[c]
+                )
             # redo mesh -- functionality based on assembly.calculateZCoords()
             mesh.append(b.p.ztop)
             b.spatialLocator = self.linked.a.spatialGrid[0, 0, ib]
 
         bounds = list(self.linked.a.spatialGrid._bounds)
         bounds[2] = array(mesh)
         self.linked.a.spatialGrid._bounds = tuple(bounds)
+        # populate remaining post-expansion criteria for conservation reports
+        self.massConservationReport["post-exp"] = (
+            list(materialMasses.values()) + [None] + list(componentMasses.values())
+        )
+        # check for differences in pre and post expansion mass greater than 1e-9 grams
+        diff = [
+            round(new - orig, 9) if new else None
+            for new, orig in zip(
+                self.massConservationReport["post-exp"],
+                self.massConservationReport["pre-exp"],
+            )
+        ]
+        self.massConservationReport["round(post - pre, 9)"] = diff
+        if any(diff):
+            # log the high level mass conservation report
+            runLog.extra(
+                tabulate(self.massConservationReport, headers="keys", floatfmt="0.8e")
+            )
+            # log the detailed mass conservation report
+            self.detailedMassConservationReport = (
+                self._setOrderForDetailedMassConservationReport()
+            )
+            runLog.debug(
+                tabulate(
+                    self.detailedMassConservationReport, headers="keys", floatfmt="0.8e"
+                )
+            )
+
+    @staticmethod
+    def _getCompHeight(component) -> float:
+        """Get the component height, if doesn't exist, get parent block height."""
+        try:
+            compHeight = component.height
+        except AttributeError:
+            compHeight = component.parent.getHeight()
+        return compHeight
+
+    def _setOrderForDetailedMassConservationReport(self) -> dict:
+        """Return the detailed mass conservation report dictionary in a useful order."""
+        keyOrder = [
+            "Block",
+            "Component",
+            "GrowFrac",
+            "(prev) c.bottom",
+            "(prev) c.top",
+            "(prev) c.height",
+            "(post) c.bottom",
+            "(post) c.top",
+            "(post) c.height",
+            "(prev) mass",
+            "(post) mass",
+            "(post - pre) mass",
+        ]
```

Comment:
This can be a (private) module variable to avoid reinstantiating it all the time.

Reply:
Addressed https://github.com/terrapower/armi/pull/1338/commits/80444f851a16baf40e0a2f54c301c84765f8d7d5